src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvGraphicSourceTest.groovy

```diff
@@ -0,0 +1,170 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.io.source.csv
+
+import edu.ie3.datamodel.io.factory.input.graphics.LineGraphicInputEntityData
+import edu.ie3.datamodel.io.factory.input.graphics.NodeGraphicInputEntityData
+import edu.ie3.datamodel.io.source.RawGridSource
+import edu.ie3.datamodel.models.input.NodeInput
+import edu.ie3.datamodel.models.input.graphics.NodeGraphicInput
+import edu.ie3.test.common.GridTestData as gtd
+import org.locationtech.jts.geom.LineString
+import org.locationtech.jts.geom.Point
+import spock.lang.Specification
+
+class CsvGraphicSourceTest extends Specification implements CsvTestDataMeta {
+
+
+	def "A CsvGraphicSource should provide an instance of GraphicElements based on valid input data correctly"() {
+		given:
+		def typeSource = new CsvTypeSource(csvSep, typeFolderPath, fileNamingStrategy)
+		def rawGridSource = new CsvRawGridSource(csvSep, gridFolderPath, fileNamingStrategy, typeSource)
+		def csvGraphicSource = new CsvGraphicSource(csvSep, graphicsFolderPath, fileNamingStrategy, typeSource, rawGridSource)
+
+		when:
+		def graphicElementsOpt = csvGraphicSource.getGraphicElements()
+
+		then:
+		graphicElementsOpt.present
+		graphicElementsOpt.ifPresent({ graphicElements ->
+			assert (graphicElements.allEntitiesAsList().size() == 3)
+			assert (graphicElements.nodeGraphics.size() == 2)
+			assert (graphicElements.lineGraphics.size() == 1)
+		})
```

Comment:
Just a simplification I recently learned:
```suggestion
		graphicElementsOpt.ifPresent({
			assert (it.allEntitiesAsList().size() == 3)
			assert (it.nodeGraphics.size() == 2)
			assert (it.lineGraphics.size() == 1)
		})
```
Only for the sake of sharing knowledge.